src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts

```diff
@@ -201,47 +201,86 @@ export class TextSubmissionViewerComponent implements OnChanges {
         return this.matches.has(file);
     }
 
-    insertToken(text: string, token: string, position: number) {
-        // prevent negative values because slice does not handle them as we would wish
-        if (position < 0) {
-            position = 0;
+    insertMatchTokens(fileContent: string): string {
+        const matches = this.getMatchesForCurrentFile().sort((m1, m2) => {
+            if (!m1.from || !m1.to || !m2.from || !m2.to) {
```

Comment:
What if we just filter the invalid matches instead of including them in the sort?
You skip them in the loop below anyways, so to avoid incidents we could just entirely ignore them.
We can still capture an exception for such cases in the filter function.

Reply:
Great idea 👍